doc/developer-guide/developer-guide.rst

```diff
@@ -0,0 +1,93 @@
+#########################################
+LinearStage Developer Guide
+#########################################
+
+The LinearStage is a standard salobj supported CSC.
+The hardware communicates via a RS-232 Serial cable.
+
+.. _developer-guide:developer-guide:dependencies:
+
+Dependencies
+============
+
+* `SAL <https://ts-sal.lsst.io>`_ - v4.0.0
+* ts_salobj - v5.x.0
+* zaber-serial - v0.9.1
+
+.. Linking to the previous versions may also be worthwhile, depending on the CSC
+
+.. _developer-guide:developer-guide:api:
+
+LinearStage API
+=============================
+
+The content in this section is autogenerated from docstrings.
+
+.. The code below should insert the docstrings from the code.
+
+.. automodapi:: lsst.ts.LinearStage
+    :no-main-docstr:
+    :no-inheritance-diagram:
+
+
+.. _developer-guide:developer-guide:build:
+
+Build and Test
+==============
+
+.. prompt:: bash
+
+    setup -kr .
+    scons
+
+.. prompt:: bash
+
+    pip install .[dev]
+    pytest --cov lsst.ts.LinearStage -ra
+
+
+.. _developer-guide:developer-guide:usage:
+
+Usage
+=====
+
+Plug in the hardware.
+
+.. code::
+
+    run_linear_stage.py 1
+
+.. _developer-guide:developer-guide:simulator:
+
+Simulator
+=========
+
+N/A
+
+
+.. _developer-guide:developer-guide:firmware:
+
+Updating Firmware of the LinearStage
+====================================
+
+N/A
+
+
+.. _developer-guide:developer-guide:documentation:
+
+Building the Documentation
+==========================
+
```

Comment:
The documentation can be build for this CSC using the following commands. It is expected this is run from the CSC development container.